### ISSUE TITLE
No updating of the feature count in the Layers Panel when QGIS is reading an oracle point table

### ISSUE THREAD

**Issue by felixroussel** [qgis/QGIS#53822]
### What is the bug or the crash?

I have layers from Oracle with a categorized symbology.
The number of features in the oracle table will evolve from tools other than QGIS.
The feature count is turned on and shows
- the total number of features (next to the layer name) and
- the feature count for each category

![Numbers of items in](https://github.com/qgis/QGIS/assets/139474661/892db671-8736-42b4-90df-e6f53178bf57)

The feature count will not change on the panel. The only way to have the figure changing is to close the QGIS project and reopen it.

Work arounds (which force the virtual layer to update the categories' feature count) are:
- closing and re-opening the project

Still it would be nice, if the virtual layer's category feature count would update immediately upon update of the source layer.

### Steps to reproduce the issue

1) Open a layer from an other database (here Oracle).
2) Right click en the layer in the layer panel and select "show feature count"
3) Edit this layer from an other client (an other QGIS is good) : remove a feature, add a feature, edit an attribute
4) Refresh the maps
5) Observe that the feature appears on the map and the feature count is not

### Versions

QGIS version
3.22.13-Białowieża
QGIS code revision
02ff117c
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.6.0
PROJ version
9.1.0
EPSG Registry database version
v10.074 (2022-08-01)
GEOS version
3.11.1-CAPI-1.17.1
SQLite version
3.39.4
PDAL version
2.4.3
PostgreSQL client version
14.3
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows 10 Version 2009

Active Python plugins
AutoLayoutTool
1.1
SpreadsheetLayers
2.1.0
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.5
processing
2.12.99
sagaprovider
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by agiudiceandrea** [comment]
@felixroussel, could you please actually try if the issue does occur also using a currently supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule) as requested in the issue report form? QGIS 3.22 is no longer supported.

**Comment by felixroussel** [comment]
> @felixroussel, could you please actually try if the issue does occur also using a currently supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule) as requested in the issue report form? QGIS 3.22 is no longer supported.

Yes, I've tried this on 3.22.13, 3.28.9 and 3.32.1 with the same result.
I'm editing the topic with a newer version

**Comment by wburt** [comment]
Perhaps related to this issue is erroneous counts after a data provider feature filter is applied. Categories and total counts are updated to zero.
